AE-1278 Add API for deleting sivu without parent

muep opened the pull request:
As of now, an exception will be produced if the API user attempts to
delete a page that still has a parent.

muep committed:
AE-1278 Add API for deleting sivu without parent

As of now, an exception will be produced if the API user attempts to
delete a page that still has a parent.

muep commented:
Tähän on tarkoitus laajentaa joku järkevä käsittely niillekin sivuille millä on lapsia, mutta tarvinnee ensin vähän tarkistaa että onko se esim. kielletty, rekursiivinen poisto vai jonkinlainen ylempään tasoon korottaminen niille lapsisivuille.

MattiNieminen commented on etp-backend/src/main/clj/solita/etp/api/sivu.clj:
```diff
@@ -41,4 +41,13 @@
+                :responses {200 {:body nil}
+                            404 {:body schema/Str}}
+                :handler (fn [{{{:keys [id]} :path} :parameters :keys [db]}]
+                           (api-response/put-response
```
put-response on tässä ehkä vähän hassu? Jos se kuitenkin tekee mitä pitää, pitäisikö sama funkkari kopioida aliakselle delete-response tms?

muep replied:
Samaa mieltä, mutta aiemminkin `:delete`n kanssa on käytetty tuota `put-response`a. Jos vaihdetaan, niin voisi kannattaa tehdä erillisenä muutoksena.

Vissiin sekä SQL:n `UPDATE` että `DELETE` antavat niin samannäköistä tulosta, että tuo put-responsen logiikka sopii sellaisenaan molempiin.

muep replied:
Tästä slackissa keskusteltiin ja todettiin, että ei tehdä mitään lisänimiä put-responselle. Saatetaan vaihtaa kuitenkin put-responsen nimi joksikin muuksi. Jätän kuitenkin tästä PR:sta sen harjoituksen pois.

MattiNieminen commented on etp-backend/src/main/sql/solita/etp/db/sivu.sql:
```diff
@@ -21,6 +21,9 @@ where
   id = :id and
   (:paakayttaja or published);
 
+-- name: delete-sivu!
```
Täällä voisi varmaan olla myös WHERE-ehto, että sivulla ei saa lapsisivuja (subquery). Sitten servicessä voisi tarkistaa lukumäärän kuinka monta riviä muuttui, ja palauttaa sen api:iin. Muistaakseni tätä samaa ideaa käytetään jossain muuallakin.

muep replied:
Ajattelin vain ottaa foreign key ongelman kiinni ja antaa siitä bad request -vastauksen, mutta tämä voisi olla myös ihan suht simppeli reitti hoitaa se asia.

muep replied:
Nyt äskettäin pushatussa versiossa olisi `:foreign-key-violation` otettu kiinni ja sen perusteella 400 status.

muep committed:
Sivu API: Handle foreign key violations

Previously these would result in an unhandled exception and 500 as
HTTP status.
